.pre-commit-config.yaml

```diff
@@ -709,8 +710,11 @@ repos:
       - id: check-base-operator-usage
         language: pygrep
         name: Check BaseOperator[Link] other imports
-        description: Make sure BaseOperator[Link] is imported from airflow.models outside of core
-        entry: "from airflow\\.models\\.baseoperator(link)? import.* BaseOperator"
+        description: Make sure BaseOperator is imported from airflow.models and BaseOperatorLink is imported from airflow.models or airflow.sdk.* outside of core
+        entry: >
+          (from airflow\\.models\\.baseoperator import.* BaseOperator\\b|
+           from airflow\\.models import.* BaseOperatorLink\\b|
+           from airflow\\.sdk\\..* import.* BaseOperatorLink\\b)
         pass_filenames: true
         files: >
           (?x)
```

Comment:
And since there is no now difference betweeen how core and providers import BaseOperatorLink, I think we can remove the Link part of this check.

Reply:
True, let me do that